### PR TITLE
Fill out and restructure readme.

### PR DESCRIPTION
**Pull request by rbt**
Took a whack at the documentation for  #184.

I think much of what remains is half-baked.

### CODE REVIEW

File: README.pod [hunk 1]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
 
   $dbh.dispose;
 
-=head2 Fetching data
+=head2 Running Queries
```
Comment (abraxxa) [inline README.pod:99]: Maybe use `Executing queries` here?

File: README.pod [hunk 2]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+
+=head3 C<execute>
+
+For a single execution of a query you may use execute directly. This starts the query within the database and returns
+a C<StatementHandle> which will be used for [Retrieving Data](#retrieving-data).
```
Comment (abraxxa) [inline README.pod:104]: `can be used` because queries like the example right below don't return any data.

File: README.pod [hunk 3]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+
+
+Parameter binding should be used where-ever possible, even if the number of parameters is dynamic. In this case the
+number of elements for IN is dynamic. The number of C<?>'s is scaled to fit the list of items, and the list
+is provided to execute as individual items.
+
+  my @value-list = 1 .. 6;
+  my $parameter-bind-marks = @value-list.map({'?'}).join(',')  # ?,?,?,?,?,?
+  my $query = 'SELECT id FROM tab WHERE id IN (%s)'.sprintf($parameter-bind-marks);
+  $dbh.execute($query, |@value-list);
```
Comment (abraxxa) [inline README.pod:145]: Is the | a typo or some weird Raku construct?

File: README.pod [hunk 4]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+
+All database drivers support basic Raku types like Int, Rat, Str, and Buf; some databases may support additional
+complex types such as an Array of Str or Array of Int which may simplify the above example significantly. Please see
+database specific documentation for descriptions of additional type support.
```
Comment (abraxxa) [inline README.pod:149]: I'd remove the `descriptions of`.

File: README.pod [hunk 5]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+  for 1 .. 100 -> $id {
+    $dbh.execute('INSERT INTO tab (id) VALUES (?)', $id);
+  }
 
-DBIish provides nearly all the perl5 DBI fetch* method to fetch values from the C<StatementHandle> object.
-However it's recommended to use the C<row> and C<allrows> methods. They provide you typed values
+  # Efficient
+  #
+  # Uses prepare to decrease overhead on the client side; often on the server-side too as the database may only need
+  # to parse the SQL once.
```
Comment (abraxxa) [inline README.pod:164]: Make the comments regular documentation please.

File: README.pod [hunk 6]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+  my $sth = $dbh.prepare('INSERT INTO tab (id) VALUES (?)');
+  for 1 .. 100 -> $id {
+    $sth.execute($id);
+  }
 
-=head3 row
+=head2 Retrieving data
+
+DBIish provides the C<row> and C<allrows> methods to fetch values from a C<StatementHandle> object returned by execute.
+These functions provide you typed values; for example an int4 field in the database will be provided as an Integer
```
Comment (abraxxa) [inline README.pod:173]: Integer is no Raku type, did you mean Int?

File: README.pod [hunk 7]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+    }
+
+    # Or as a shorter example:
+    for $dbh.execute('SELECT id, col FROM tab').allrows(:array-of-hash) -> $row {
+       say $row<id>  # 1␤3
+    }
+
+=head3 C<dispose>
+
+After you have finished with the statement handle, you can free the memory it used immediately using C<dispose>.
```
Comment (abraxxa) [inline README.pod:212]: After you have fetched all data using the statement handle, you can free its memory immediately using C<dispose>.

File: README.pod [hunk 8]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+
+    $sth.dispose
```
Comment (abraxxa) [inline README.pod:214]: Missing ;

File: README.pod [hunk 9]
```diff
@@ -98,40 +96,171 @@ C<dispose> method:
...
+
+Database Driver name for the connection
+
+=item C<native-message>
+
+Unmodified message received from the database server.
+
+=item C<code>
+
+Int Return code from the local client library for the call; typically -1. This is not an SQL State.
```
Comment (abraxxa) [inline README.pod:240]: lowercase `return` and `state`.

File: README.pod [hunk 10]
```diff
@@ -169,47 +299,64 @@ Currently the following backends are included:
...
+Pg arrays are supported for both writing via execute and retrieval via C<row/allrows>.
+You will get the properly typed array according to the field type.
 
 Passing an array to C<execute> is now implemented. But you can also use the
 C<pg-array-str> method on your Pg StatementHandle to convert an Array to a
 string Pg can understand:
 
   # Insert an array via an execute statement
-  my $sth = $db.execute('INSERT INTO tab (array_column) VALUES ($1);', @data);
+  my $sth = $dbg.execute('INSERT INTO tab (array_column) VALUES ($1);', @data);
```
Comment (abraxxa) [inline README.pod:333]: Should be `$dbh`

Comment (rbt) [reply README.pod]: Thank-you for the review. I've cleaned these up as directed.

File: README.pod (hunk 3 above)
Comment (rbt) [reply README.pod:145]: |@ is the opposite of a slurpy *@.

my @params = 1,2,3;
foo(|@params) is foo(1,2,3) rather than foo( [1,2,3] )

**Pull request merged**